src/loki/reach.cc

```diff
@@ -5,34 +5,42 @@ using namespace valhalla::baldr;
 namespace valhalla {
 namespace loki {
 
-directed_reach SimpleReach(const DirectedEdge* edge,
-                           uint32_t max_reach,
-                           GraphReader& reader,
-                           const sif::EdgeFilter& edge_filter,
-                           const sif::NodeFilter& node_filter,
-                           uint8_t direction) {
-
-  // no reach to start with
+Reach::Reach() : Dijkstras() {
+  // Mock up the Location struct with the important stuff missing
+  auto* path_edge = locations_.Add()->add_path_edges();
+  path_edge->set_distance(0);
+  path_edge->set_begin_node(false);
+  path_edge->set_end_node(false);
+}
+
+directed_reach Reach::operator()(const DirectedEdge* edge,
+                                 const baldr::GraphId edge_id,
+                                 uint32_t max_reach,
+                                 GraphReader& reader,
+                                 const std::shared_ptr<sif::DynamicCost>& costing,
+                                 uint8_t direction) {
+
+  // no reach is needed
   directed_reach reach{};
   if (max_reach == 0)
     return reach;
+  max_reach_ = max_reach;
+
+  auto node_filter = costing->GetNodeFilter();
+  auto edge_filter = costing->GetEdgeFilter();
 
-  // TODO: throw these vectors into a cache that we reuse
   // we keep a queue of nodes to expand from, to prevent duplicate expansion we use a set
   // each node we pop from the set will increase the reach and be added to the done set
   // the done set is used to avoid duplicate expansion of already dequeued nodes
   // we also track how many nodes were added as transitions from other levels
   // this allows us to have "duplicate" nodes but not do any trickery with the expansion
-  std::unordered_set<uint64_t> queue, done;
-  queue.reserve(max_reach);
-  done.reserve(max_reach);
-  size_t transitions = 0;
 
   // helper lambda to enqueue a node
+  Clear();
```

Comment:
makes sure we have spce to do the expansion in our various member variables can clears them from previous runs